packages/config/scripts/checkVerifiedContracts/tasks.ts

```diff
@@ -0,0 +1,54 @@
+import { EtherscanClient, Logger, TaskQueue } from '@l2beat/common'
+import { EthereumAddress } from '@l2beat/types'
+
+import { isContractVerified } from './etherscan'
+import { VerificationMap } from './output'
+
+export async function verifyContracts(
+  addresses: EthereumAddress[],
+  previouslyVerified: Set<EthereumAddress>,
+  etherscanClient: EtherscanClient,
+  workersCount: number,
+  logger: Logger,
+): Promise<VerificationMap> {
+  logger.info(`Processing ${addresses.length} addresses.`)
+  const result: VerificationMap = {}
+  let errorOccured = false
+
+  // Copy previously verified contracts directly to result
+  addresses
+    .filter((a) => previouslyVerified.has(a))
+    .forEach((a) => (result[a.toString()] = true))
+
+  const taskQueue = new TaskQueue(
```

Comment:
IMO this could be extracted from here to something like PromiseAllThrotthled(?) but with a limited parallelism. It could be just a separate function in this file for now but we also could move it to a @common lib.

Reply:
I tried to do that, but taskQueue has different semantics than Promise.all, because taskQueue is fed *data*, while Promise.all is fed *promises*. I'd have to implement something like `PromiseAllThrottled()` from scratch, so I decided to stick to what we already use.

Saying that, I have just refactored this call to a separate function - it does look cleaner.